staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -508,21 +508,14 @@ func (c *Cacher) Watch(ctx context.Context, key string, opts storage.ListOptions
 	// underlying watchCache is calling processEvent under its lock.
 	c.watchCache.RLock()
 	defer c.watchCache.RUnlock()
-	initEvents, err := c.watchCache.GetAllEventsSinceThreadUnsafe(watchRV)
+	cacheInterval, err := c.watchCache.GetAllEventsSinceThreadUnsafe(watchRV)
 	if err != nil {
 		// To match the uncached watch implementation, once we have passed authn/authz/admission,
 		// and successfully parsed a resource version, other errors must fail with a watch event of type ERROR,
 		// rather than a directly returned error.
 		return newErrWatcher(err), nil
 	}
 
-	// With some events already sent, update resourceVersion so that
-	// events that were buffered and not yet processed won't be delivered
-	// to this watcher second time causing going back in time.
-	if len(initEvents) > 0 {
-		watchRV = initEvents[len(initEvents)-1].ResourceVersion
-	}
```

Comment:
We can't just remove it.

It seems that GetAllEventsSinceThreadUnsafe should basically return:
- cacheInterval
- current resource version (and we will simply use that here)

Reply:
FWIW - returning RV from it is a simple change so let's maybe do that now in a separate PR.

Reply:
OK - I see that you're addressing it below differently. That sounds fine too.
But as I mentioned - let's maybe move it to a separate PR.